spec/03-plugins/18-acl/01-api_spec.lua

```diff
@@ -152,10 +153,114 @@ for _, strategy in helpers.each_strategy() do
+
+          local res  = assert(admin_client:post("/consumers/anna/acls", {
+            body = {
+              group = "foo"
+            },
+            headers = {
+              ["Content-Type"] = "application/json"
+            },
+          }))
+          local body = assert.res_status(201, res)
+          local ag   = cjson.decode(body)
+
+          local res  = assert(admin_client:post("/consumers/jack/acls", {
+            body = {
+              group = "foo"
+            },
+            headers = {
+              ["Content-Type"] = "application/json"
+            },
+          }))
+          local body = assert.res_status(201, res)
+          local jg   = cjson.decode(body)
+
+          local res  = assert(admin_client:get("/consumers/anna/acls/foo"))
+          local body = assert.res_status(200, res)
+          local ag2  = cjson.decode(body)
+
+          local res  = assert(admin_client:get("/consumers/jack/acls/foo"))
+          local body = assert.res_status(200, res)
+          local jg2  = cjson.decode(body)
+
+          assert.same(ag, ag2)
+          assert.same(jg, jg2)
+          assert.not_same(jg, ag)
+          assert.not_same(jg2, ag2)
+          assert.not_same(jg, ag2)
+          assert.not_same(jg2, ag)
+
+          local res  = assert(admin_client:delete("/consumers/anna"))
+          local _    = assert.res_status(204, res)
+
+          local res  = assert(admin_client:delete("/consumers/jack"))
+          local _    = assert.res_status(204, res)
+        end)
+        it("retrieves right ACL by group when multiple consumers share the same group name created with PUT", function()
+          local res  = assert(admin_client:put("/consumers/anna"))
+          local body = assert.res_status(200, res)
+          local _    = cjson.decode(body)
+
+          local res  = assert(admin_client:put("/consumers/jack"))
+          local body = assert.res_status(200, res)
+          local _    = cjson.decode(body)
+
+          local res  = assert(admin_client:put("/consumers/anna/acls/foo"))
+          local body = assert.res_status(200, res)
+          local ag   = cjson.decode(body)
+
+          local res  = assert(admin_client:put("/consumers/jack/acls/foo"))
+          local body = assert.res_status(200, res)
+          local jg   = cjson.decode(body)
+
+          local res  = assert(admin_client:get("/consumers/anna/acls/foo"))
+          local body = assert.res_status(200, res)
+          local ag2  = cjson.decode(body)
+
+          local res  = assert(admin_client:get("/consumers/jack/acls/foo"))
+          local body = assert.res_status(200, res)
+          local jg2  = cjson.decode(body)
+
+          assert.same(ag, ag2)
+          assert.same(jg, jg2)
+          assert.not_same(jg, ag)
+          assert.not_same(jg2, ag2)
+          assert.not_same(jg, ag2)
+          assert.not_same(jg2, ag)
+
+          local res  = assert(admin_client:delete("/consumers/anna"))
+          local _    = assert.res_status(204, res)
```

Comment:
I think assigning this to the unnamed variable serves no purpose.

```suggestion
          assert.res_status(204, res)
```